### PR TITLE
Use flash messages on application page

### PR DESCRIPTION
**Pull request by luontola**
#1469

# Definition of Done / Review checklist

## Reviewability
- [x] link to issue

## Documentation
- [x] update changelog if necessary
- [x] add or update docstrings for namespaces and functions

## Testing
- [x] complex logic is unit tested
- [x] valuable features are integration / browser / acceptance tested automatically

## Accessibility
- [x] all icons have the aria-label attribute
- [x] all fields have a label
- [x] errors are linked to fields with aria-describedby
- [x] contrast is checked
- [x] conscious decision about where to move focus after an action

## Follow-up
- [x] new tasks are created for pending or remaining tasks
- [x] no critical TODOs left to implement

### CODE REVIEW

File: src/cljs/rems/application.cljs [hunk 1]
```diff
@@ -87,7 +87,7 @@
 (rf/reg-event-fx
  ::enter-application-page
  (fn [{:keys [db]} [_ id]]
-   {:db (dissoc db ::application ::edit-application)
+   {:db (dissoc db ::application ::edit-application ::flash-message/message)
```
Comment (opqdonut) [inline src/cljs/rems/application.cljs]: hmm, clearing somebody else's "private" state feels a bit iffy

File: src/cljs/rems/flash_message.cljs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  (:require [clojure.string :as str]
+            [re-frame.core :as rf]
+            [rems.atoms :as atoms]))
+
+(rf/reg-sub ::message (fn [db _] (::message db)))
+
+(rf/reg-event-fx
+ ::show-flash-message
+ (fn [{:keys [db]} [_ message]]
+   (.scrollTo js/window 0 0)
```
Comment (opqdonut) [inline src/cljs/rems/flash_message.cljs]: shouldn't we scroll to the flash message component?

**Comment by okahilak** [comment]
Most of the flash messages seem to work nicely! I noticed one that had a slight visual glitch. When submitting an application that has an empty field, there is first a green "success" flash message, which is quickly replaced by the "error" flash message.

I also noticed that "Close Application" does not work properly -- it is left open after closing, which allows another "Close" operation, which causes an error. However, it turns out that this patch did not introduce the problem, but it exists already in master, so I created another issue for that: https://github.com/CSCfi/rems/issues/1562

**Comment by luontola** [comment]
That fixes the glitch.

**Comment by okahilak** [comment]
> That fixes the glitch.

Seems to work now, and the code change looks fine. Thanks.

**Pull request merged**